### PR TITLE
added basic box plot and caching for plot

### PR DESCRIPTION
**Pull request by carsonwolber**
## Description

<what this pull request is doing>

- this PR sets up most the bulk work for a box plot visualizing similar users data as well as some basic interactive features
- caches the currently non-functional checkboxes and svg itself
- renames and refactors the structure of the old `Chart.jsx` file to be more specific now that there are two charts

<what will be done in later PRs and not included here>

- The next PR will finish out this plot, namely by applying the interactive features the checkboxes suggest

## Milestones
This improves on TC 2 milestones: 

- Users of EconMetrics want to be able to visualize their financial information and their information compared to other users. 
- They also want to be able to customize/filter what information is pooled into the data visualized. 
- In order for this visualization to be performant and responsive, we will have to implement a caching system

## Resources

- [this](https://d3js.org/d3-scale-chromatic/categorical#schemeTableau10) helped me with the dot color schemes
- [this](https://mui.com/material-ui/react-checkbox/) documentation is for the checkbox components
- [this](https://developer.mozilla.org/en-US/docs/Web/SVG/Tutorial/Getting_Started) helped again with referencing general SVG dom manipulation 
- [this](https://observablehq.com/@mukhtyar/box-plot) example inspired the general UI/look of my svg
- [this](https://d3-graph-gallery.com/graph/boxplot_show_individual_points.html) was another reference for D3 SVG graphs

## Test Plan
This is a video showing off the current graph, similar-users interactive features (hover and click), and caching of checkbox components and the new graph

https://github.com/user-attachments/assets/0a903c79-e648-456a-b7e7-71e793d07015

### CODE REVIEW

File: frontend/src/main.jsx [hunk 1]
```diff
@@ -4,7 +4,5 @@ import App from './App.jsx'
 import './index.css'
 
 ReactDOM.createRoot(document.getElementById('root')).render(
-  <React.StrictMode>
     <App />
```
Comment (carsonwolber) [inline frontend/src/main.jsx:7]: I removed `React.StrictMode` because it was causing components to render twice, this was fine before but since in this new PR I'm caching boolean values it was leading to unexpected behavior switching the values. Does anyone foresee any issues that may arise from this or should this be fine?

**Comment by William-La** [comment]
To clarify, this PR only contains the components of the box plot? Will a follow up PR have the rest?

**Comment by William-La** [comment]
nit: The "Show your stats" and "Show box plot" buttons are close to both graphs in the video above, which is a bit confusing.  A future point for cleanup could make the options for the graphs more distinguishable

File: frontend/src/ProfilePage/Graphs/CompBoxPlot.jsx [hunk 2]
```diff
@@ -0,0 +1,167 @@
+import { useEffect, useRef } from 'react';
```
Comment (William-La) [inline frontend/src/ProfilePage/Graphs/CompBoxPlot.jsx:1]: at first glance, the CompBoxPlot and TimeChart graphs have a similar implementation approach where you do some computations and then construct a svg. How much of this code is re-usable? Could we refactor portions like the svg creation?

**Comment by carsonwolber** [comment]
> To clarify, this PR only contains the components of the box plot? Will a follow up PR have the rest?

This PR is the basic foundation of the box plot– the PR I'm about to make adds more advanced interactive features (primarily implementing what the checkboxes suggest) which includes the element for the actual box in addition to a new class for user dots.

**Comment by carsonwolber** [comment]
> nit: The "Show your stats" and "Show box plot" buttons are close to both graphs in the video above, which is a bit confusing. A future point for cleanup could make the options for the graphs more distinguishable

I added this tooltip feature in between the graphs in the PR I'm about to launch which I thought gave more of a clear transition between the two graphs 

Screenshot 2024-07-22 at 4 21 46 PM

File: frontend/src/ProfilePage/Graphs/CompBoxPlot.jsx (hunk 2 above)
Comment (carsonwolber) [reply frontend/src/ProfilePage/Graphs/CompBoxPlot.jsx:1]: yeah so the values that are easily repeated are being stored in the `chartConfig` JSON, the big thing they both share from here is margins. In terms of the code to create the graphs, there's not a ton of 1:1 overlap between the two and in general it'd be hard to have a generic `graph` component create different graphs based on props if that's what you're suggesting.  Here for instance, the creation and scaling of both axes is pretty similar, but it's not a perfect match. Especially for the x-axes. The earlier graph is more of a conventional cartesian graph while the one I'm working on now having x-ticks represent categories that requires entirely different padding to control how the groupings are formatted. The differences in graphs also motivates other small differences in formatting (e.g. in the box-plot graph, width of the svg itself is 1/3 wider than the old graph, the x axis is farther down to give more room to the more vertical data formatting of this graph) 

TL;DR: while the general structure of svg creation is pretty similar across graphs(in respect to creating the blank plane itself), there are subtle differences that would  in my opinion make creating the svgs generically more trouble than it's worth at the scale of 2 graphs

Comment (pangers) [reply frontend/src/ProfilePage/Graphs/CompBoxPlot.jsx:1]: Thanks for the explanation, that makes sense

File: frontend/src/main.jsx (hunk 1 above)
Comment (pangers) [reply frontend/src/main.jsx:7]: Its probably fine

**Review by pangers — APPROVED** [review]
